bunq/sdk/model/core/bunq_model.py

```diff
@@ -0,0 +1,180 @@
+from bunq import Pagination
+from bunq.sdk.http import api_client
+from bunq.sdk.http.bunq_response import BunqResponse
+from bunq.sdk.json import converter
+
+
+class BunqModel(object):
+    # Field constants
+    _FIELD_RESPONSE = 'Response'
+    _FIELD_PAGINATION = 'Pagination'
+    _FIELD_ID = 'Id'
+    _FIELD_UUID = 'Uuid'
+
+    # The very first index of an array
+    _INDEX_FIRST = 0
+
+    __STRING_FORMAT_EMPTY = ''
+    __STRING_FORMAT_FIELD_FOR_REQUEST_ONE_UNDERSCORE = '_field_for_request'
+    __STRING_FORMAT_FIELD_FOR_REQUEST_TWO_UNDERSCORE = '__field_for_request'
+
+    def is_all_field_none(self):
+        raise NotImplementedError
+
+    def to_json(self):
+        """
+        :rtype: str
+        """
+
+        return converter.class_to_json(self)
+
+    @staticmethod
+    def from_json(json_str):
+        raise NotImplementedError
+
+    @classmethod
+    def _from_json_array_nested(cls, response_raw):
+        """
+        :type response_raw: api_client.BunqResponseRaw
+
+        :rtype: bunq.sdk.client.BunqResponse[cls]
```

Comment:
why the `[cls]`

Reply:
I am honestly not sure. What do you suggest?